packages/services/api/src/modules/billing/resolvers/Mutation/upgradeToPro.ts

```diff
@@ -36,6 +37,25 @@ export const upgradeToPro: NonNullable<MutationResolvers['upgradeToPro']> = asyn
           operations: Math.floor(args.input.monthlyLimits.operations / 1_000_000),
         },
       });
+
+      const currentUser = await injector.get(AuthManager).getCurrentUser();
+      injector.get(AuditLogManager).createLogAuditEvent(
+        {
+          eventType: 'SUBSCRIPTION_CREATED',
+          subscriptionCreatedAuditLogSchema: {
+            operations: args.input.monthlyLimits.operations,
+            paymentMethodId: args.input.paymentMethodId,
+            newPlan: 'PRO',
+            previousPlan: 'HOBBY',
+          },
+        },
+        {
+          organizationId: organizationId,
+          userEmail: currentUser.email,
+          userId: currentUser.id,
+          user: currentUser,
+        },
+      );
```

Comment:
These calls should happen within the business logic (BillingProvider), not the resolvers.

Reply:
Hmm @n1ru4l it's a bit tricky, no? 

Most of these providers are defined as `Singleton` scope, and I think this will prevent the provider from accessing other `Operation` scope providers? (for example, to get the current user)
(and alternative could be to change AuditLogProvider to be `Operation`?)

Reply:
For example, I need to add this for each provider:

```
    private authManager: AuthManager,
    private auditLogManager: AuditLogManager,
```
```

@Injectable({
  global: true,
  scope: Scope.Singleton,
})
export class BillingProvider {
  private logger: Logger;
  private billingService;


  enabled = false;

  constructor(
    logger: Logger,
    private storage: Storage,
    private authManager: AuthManager,
    private auditLogManager: AuditLogManager,
    @Inject(BILLING_CONFIG) billingConfig: BillingConfig,
  ) {
```

Reply:
We can make these non-singletons. I really want to avoid growing the resolvers with business logic. If we introduce a public API, that does not use GraphQL we have to duplicate all the audit log inserts.

Reply:
I think most of the *Manager classes are already Operation scoped, so it won't be an big issue. I agree with Laurin's take.
We may have a new GraphQL API or even OpenAPI in future and keeping things in resolvers means we need to move it anyway.

Reply:
Also, when we do it in resolvers, the audit log needs to run after the business logic ran, otherwise we log something that did not happen